src/components/JobsSearchInput/index.tsx

```diff
@@ -0,0 +1,103 @@
+import { ChangeEvent, MouseEventHandler, useEffect, useRef, useState } from "react";
+import DropdownArrow from "../../../public/images/dropdownArrow.svg";
+
+interface JobsSearchInput {
+  type: 'select' | 'text';
+  label: string;
+  options?: string[];
+  placeholder?: string;
+}
+
+const JobsSearchInput = ({ type, label, options, placeholder }: JobsSearchInput) => {
+  const select = useRef(null);
+  const [selectedData, setSelectedData] = useState('Escolha');
+  const [isOpenDropdown, setIsOpenDropdown] = useState(false);
+  const [isUserAction, setIsUserAction] = useState(false);
+  const [inputData, setInputData] = useState('');
+
+  useEffect(() => {
+    if (!isOpenDropdown) return;
+    const closeDropdownHandle = () => setIsOpenDropdown(false);
+    if (window) {
+      window.addEventListener('click', closeDropdownHandle);
+      return () => window.removeEventListener('click', closeDropdownHandle);
+    };
+  }, [isOpenDropdown]);
+
+  const dropdownHandle = () => setIsOpenDropdown(!isOpenDropdown);
+
+  const clickOptionHandle: MouseEventHandler<HTMLDivElement> = (event) => {
+    const clickedData = event.currentTarget.outerText;
+    const selectLength = select.current.options.length;
+    for (let i = 0; i < selectLength; i++) {
+      const optionValue = select.current.options[i].value;
+      if (clickedData === optionValue) {
+        select.current.value = optionValue;
+        setSelectedData(optionValue);
+      };
+    };
+    setIsUserAction(true);
+  };
+
+  const changeHandle = (event: ChangeEvent<HTMLInputElement>) => {
+    setInputData(event.target.value);
+    if (event.target.value === '') setIsUserAction(false);
+    if (event.target.value !== '') setIsUserAction(true);
+  };
```

Comment:
Acredito que aqui seria melhor escrito se fosse dessa forma, claro teria que ver no quesito render do React.

```suggestion
 const changeHandle = (event: ChangeEvent<HTMLInputElement>) => {
    setInputData(event.target.value);
    setUserAction(event.target.value  === '' ? false : true)
  };
```

Reply:
Obrigado pela sugestão, @Alex-dll . Eu fiz a alteração. Vou resolver a conversa.